source/data-channel.js

```diff
@@ -180,6 +180,53 @@ Skylink.prototype._createDataChannel = function(peerId, dataChannel, bufferThres
   }
 };
 
+/**
+ * Function that refreshes the main messaging Datachannel.
+ * @method refreshDatachannel
+ * @param {String} [peerId] The target Peer ID to retrieve connection stats from.
+ * @example
+ *   // Example 1: Retrieve offerer and refresh datachannel:
+ *   skylink.on("dataChannelState", function (state, peerId, error, channelName, channelType) {
+ *   if (channelType === skylink.DATA_CHANNEL_TYPE.MESSAGING &&
+ *    state === skylink.DATA_CHANNEL_STATE.CLOSED) {
+ *    var userWeight = skylink.getPeerInfo().config.priorityWeight;
+ *    var peerWeight = skylink.getPeerInfo(peerId).config.priorityWeight;
+ *    // Determine who is offerer because as per SM protocol, higher weight is offerer
+ *    if (userWeight > peerWeight) {
+ *      skylink.refreshDatachannel(peerId);
+ *    }
+ *  }
+ *  });
+ * @for Skylink
+ * @since 0.6.30
+ */
+
+Skylink.prototype.refreshDatachannel = function (peerId) {
+
+  var self = this;
+  if(!self._dataChannels[peerId]) {
```

Comment:
@geekavi `!self._dataChannels[peerId]` means if datachannel does not exist, so it means we are refreshing when datachannel does not exist. Perhaps we do `if (self._dataChannels[peerId] && self._dataChannels[peerId].main)`?